### PR TITLE
Custom colored

### PR DESCRIPTION
**Pull request by shadowv0vshadow**
Define a "z" op, for custom color outputs.

Payload like this:
```
    operation_found_at_inputs["payload"] = {
        "9527efa43262636d8f5917fc763fbdd09333e4b387afd6d4ed7a905a127b27b4i0": {
            "0": 200,
            "1": 300,
        },
        "8888e25790cc118773c8590522e1cab2ccc073a9375b238aaf9aadb13a764a13i0": {
            "2": 8000,
            "4": 8000,
        }
    }
``` 
We can set any int in every outputs. and custom colored ft.
Be careful, If your payload is flawed, it may lead to unnecessary burned.

### CODE REVIEW

File: electrumx/lib/atomicals_blueprint_builder.py [hunk 1]
```diff
@@ -325,8 +326,37 @@ def calculate_output_blueprint_fts(cls, tx, ft_atomicals, operations_found_at_in
...
+    for atomical_id, atomical_info in sorted(ft_atomicals.items()):
+      remaining_value = atomical_info.total_atomical_value
+      for out_idx, txout in enumerate(tx.outputs):
+        expected_output_index = str(out_idx)
+        compact_atomical_id = location_id_bytes_to_compact(atomical_id)
+        expected_value = operations_found_at_inputs["payload"][compact_atomical_id].get(expected_output_index, 0)
+        if expected_value <= 0 or remaining_value <= 0:
+            continue
+        if expected_value > txout.value:
+            expected_value = txout.value
```
Comment (atomicals) [inline electrumx/lib/atomicals_blueprint_builder.py:350]: This results in a burn correct?

File: electrumx/lib/atomicals_blueprint_builder.py [hunk 2]
```diff
@@ -325,8 +326,37 @@ def calculate_output_blueprint_fts(cls, tx, ft_atomicals, operations_found_at_in
...
+  
+  @classmethod
+  def custom_color_ft_atomicals(cls, ft_atomicals, operations_found_at_inputs, tx):
+    output_colored_map = {}
+    fts_burned = {}
+    cleanly_assigned = True
+    for atomical_id, atomical_info in sorted(ft_atomicals.items()):
+      remaining_value = atomical_info.total_atomical_value
+      for out_idx, txout in enumerate(tx.outputs):
+        expected_output_index = str(out_idx)
```
Comment (atomicals) [inline electrumx/lib/atomicals_blueprint_builder.py:344]: Should we check that `isinstance(out_idx, int)` is true first?

**Review by atomicals — CHANGES_REQUESTED** [review]

File: electrumx/lib/atomicals_blueprint_builder.py (hunk 1 above)
Comment (shadowv0vshadow) [reply electrumx/lib/atomicals_blueprint_builder.py:353]: If the value to be colored in the payload exceeds the actual Satoshis contained, only the Satoshis actually contained in this UTXO will be colored, and the burn amount will also be calculated correctly.

Comment (atomicals) [reply electrumx/lib/atomicals_blueprint_builder.py:353]: LGTM

**Review by atomicals — APPROVED** [review]
LGTM

File: tests/lib/test_atomicals_blueprint_builder.py [hunk 3]
```diff
@@ -898,4 +899,85 @@ def mock_mint_fetcher(self, atomical_id):
...
+    raw_tx = bytes.fromhex(raw_tx_str)
+    subject_atomical_id = b'\x13Jv:\xb1\xad\x9a\xaf\x8a#[7\xa9s\xc0\xcc\xb2\xca\xe1"\x05Y\xc8s\x87\x11\xcc\x90W\xe2\x88\x88\x00\x00\x00\x00'
+    subject_atomical_id1 = b"\xb4'{\x12Z\x90z\xed\xd4\xd6\xaf\x87\xb3\xe43\x93\xd0\xbd?v\xfc\x17Y\x8fmcb2\xa4\xef'\x95\x00\x00\x00\x00"
+    tx, tx_hash = coin.DESERIALIZER(raw_tx, 0).read_tx_and_hash()
+    # print(hash_to_hex_str(subject_atomical_id))
+    # print(hash_to_hex_str(subject_atomical_id1))
+    operation_found_at_inputs = parse_protocols_operations_from_witness_array(tx, tx_hash, True)
+    # z means costom color
```
Comment (siriusye) [inline tests/lib/test_atomicals_blueprint_builder.py:913]: typo

**Comment by ChanHongMing** [comment]
may I know what will happen if the user input float number in the payload's utxo?

**Pull request merged**